rasterio/_io.pyx

```diff
@@ -1845,27 +1845,27 @@ cdef class InMemoryRaster:
         cdef int i = 0  # avoids Cython warning in for loop below
         cdef const char *srcwkt = NULL
         cdef void *osr = NULL
+        cdef void *memdriver = NULL
 
         # Several GDAL operations require the array of band IDs as input
         self.band_ids[0] = 1
 
-        cdef void *memdriver = _gdal.GDALGetDriverByName("MEM")
-        if memdriver == NULL:
-            raise DriverRegistrationError(
-                "MEM driver is not registered.")
-
-        self.dataset = _gdal.GDALCreate(
-            memdriver,
-            "output",
-            image.shape[1],
-            image.shape[0],
-            1,
-            <_gdal.GDALDataType>dtypes.dtype_rev[image.dtype.name],
-            NULL
-        )
-
-        if self.dataset == NULL:
-            raise ValueError("NULL output datasource")
+        try:
+            with CPLErrors() as cple:
+                memdriver = _gdal.GDALGetDriverByName("MEM")
+                cple.check()
+        except:
+            raise DriverRegistrationError("MEM driver is not registered.")
+
+        try:
+            with CPLErrors() as cple:
+                self.dataset = _gdal.GDALCreate(
+                    memdriver, "output", image.shape[1], image.shape[0],
+                    1, <_gdal.GDALDataType>dtypes.dtype_rev[image.dtype.name],
+                    NULL)
+                cple.check()
+        except:
+            raise
 
```

Comment:
Why the try/except:raise here (and a few other places)? What would happen if we just used the `CPLErrors` context manager?